assets/js/translator.js

```diff
@@ -1018,6 +994,36 @@ function showTranslateWebpageInterface(url /*: ?string */, ignoreIfEmpty /*: ?bo
     });
 }
 
+function updateDetect(state /*: bool */) {
+    var oldState;
+
+    if(state) {
+        oldState = $('#detect').hasClass('activeAfterCancel');
+
+        if(oldState) {
+            $('#detect').removeClass('activeAfterCancel');
+            $('#detect').click();
```

Comment:
This is problematic. Consider the following sequence of events:

1. I type some text in
2. I click detect (it detects Spanish)
3. I click website translation (it switches to Spanish in the recent languages NOT detect and disables the detect button)
4. I cancel website translation and go back to text translation

The current code would click the detect button and a new request would be sent to APy. The old code doesn't do that since we already know what the detected language is and don't need a new request, hence the call to `detectLanguageSuccess...` only.

Reply:
Okay, not I get it. Thanks for the explanation.